libs/dal/src/repositories/topic/topic.entity.ts

```diff
@@ -0,0 +1,18 @@
+import { Types } from 'mongoose';
+
+type EnvironmentId = Types.ObjectId;
+type OrganizationId = Types.ObjectId;
+type TopicId = Types.ObjectId;
+type TopicKey = string;
+type UserId = Types.ObjectId;
```

Comment:
do you plan to reuse these types? if not I don't see the reason defining them separatelly

Reply:
Yes, I would like to slowly move them to the shared of DAL and start replacing primitive types for this kind of defined type wrappers for a more robust typing. I always find them useful to avoid argument mixing in the domain.